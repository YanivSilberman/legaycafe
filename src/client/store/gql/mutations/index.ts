export { default as createMessageGql } from "./createMessage";
export { default as loginUserGql } from "./loginUser";
export { default as toggleUserTypingGql } from "./toggleUserTyping";
