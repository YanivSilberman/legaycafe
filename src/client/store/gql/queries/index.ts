export { default as allUsersGql } from "./allUsers";
export { default as messagesGql } from "./messages";
export { default as userGql } from "./user";
export { default as usersTypingGql } from "./usersTyping";
