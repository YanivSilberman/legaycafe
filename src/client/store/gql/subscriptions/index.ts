export { default as newMessageSubscriptionGql } from "./newMessageSubscription";
export { default as userTypingSubscriptionGql } from "./userTypingSubscription";
