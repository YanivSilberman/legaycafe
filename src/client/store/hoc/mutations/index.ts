export { default as withCreateMessage } from "./withCreateMessage";
export { default as withLoginUser } from "./withLoginUser";
export { default as withToggleTyping } from "./withToggleTyping";
