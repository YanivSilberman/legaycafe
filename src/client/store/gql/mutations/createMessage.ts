import gql from 'graphql-tag';

export default gql`
  mutation createMessage($userId: ID! $text: String! $chat: ID!) {
    createMessage(
      userId: $userId,
      text: $text,
      chat: $chat
    ) {
      _id
      text
    }
  }
`;
