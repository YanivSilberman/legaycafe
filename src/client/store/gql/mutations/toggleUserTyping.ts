import gql from 'graphql-tag';

export default gql`
  mutation toggleUserTyping($_id: ID!, $isTyping: Boolean!) {
    toggleUserTyping(_id: $_id, isTyping: $isTyping)
  }
`;
