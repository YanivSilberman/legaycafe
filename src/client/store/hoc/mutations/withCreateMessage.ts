import { graphql } from 'react-apollo';
import { createMessageGql } from '../../gql/mutations';

export default graphql(createMessageGql, {
  name: "createMessage",
  props: ({ ownProps: { chat }, createMessage }:any) => ({
    createMessageMutation: (variables:object, cb:any) =>
      createMessage({ variables: { ...variables, chat } }).then(() => {
        cb();
      })
    })
})
