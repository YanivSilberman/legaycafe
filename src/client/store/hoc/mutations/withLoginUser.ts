import { graphql } from 'react-apollo';
import { loginUserGql } from '../../gql/mutations';

export default graphql(loginUserGql, {
  name: "loginUser",
  props: ({ loginUser }) => ({
    loginUserMutation: variables => loginUser({ variables })
  })
})
