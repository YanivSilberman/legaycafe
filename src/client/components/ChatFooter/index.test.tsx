import 'jsdom-global/register';
import * as React from 'react';
import { EditorState, ContentState, convertToRaw } from 'draft-js';
import { MockedProvider } from 'react-apollo/test-utils';
import * as ReactRenderer from 'react-test-renderer';
import * as ReactShallowRenderer from 'react-test-renderer/shallow';
import * as chai from 'chai';
import { shallow, mount } from 'enzyme';
import sinon from 'sinon';
import jsdom from 'jsdom';
import wait from 'waait';

import Component from './index';
import { createMessageGql, toggleUserTypingGql } from '../../store/gql/mutations';

const defaultProps = {
  isEmpty: false,
  userId:"userid",
  toggleUserTypingMutation: () => null,
  classes: {},
  chat: "chat123"
}

const editorState = EditorState.createWithContent(ContentState.createFromText('Hello'))

const mocks = [
  {
    request: {
      query: createMessageGql,
      variables: {
        userId: "userid",
        text: JSON.stringify(convertToRaw(editorState.getCurrentContent())),
        chat: "chat123"
      },
    },
    result: {
      data: {
        createMessage: {
          __typename: "Message",
          _id: 'msg123',
          text: JSON.stringify(convertToRaw(editorState.getCurrentContent()))
        },
      },
    },
  }
];


describe('ChatFooter component test',() => {
  it('component loads', () => {
    const renderer = ReactShallowRenderer.createRenderer();

    expect(renderer.render(
      <Component {...defaultProps} />
    )).toMatchSnapshot()
  })

  it('Disabled on click', async () => {
    const wrapper = mount((
      <MockedProvider mocks={mocks}>
        <Component {...defaultProps} />
      </MockedProvider>
    ));

    wrapper.setState({
      editorState: EditorState.createWithContent(ContentState.createFromText('Hello'))
    });

    await wait(0);
    wrapper.find('button').simulate('click');

    await wait(0);
    chai.expect(wrapper.find('button').props().disabled).to.equal(true);
  })
});
