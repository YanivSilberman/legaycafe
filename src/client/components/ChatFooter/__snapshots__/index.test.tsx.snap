// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatFooter component test component loads 1`] = `
<Apollo(ChatFooter)
  chat="chat123"
  classes={
    Object {
      "button": "Apollo(ChatFooter)-button-2",
      "chatFooter": "Apollo(ChatFooter)-chatFooter-1",
      "rightIcon": "Apollo(ChatFooter)-rightIcon-3",
    }
  }
  isEmpty={false}
  toggleUserTypingMutation={[Function]}
  userId="userid"
/>
`;
